go/rtl/auth.go

```diff
@@ -118,6 +118,10 @@ func (s *AuthSession) Do(result interface{}, method, ver, path string, reqPars m
 		return fmt.Errorf("response error. status=%s. error=%s", res.Status, string(b))
 	}
 
+	if res.StatusCode == 204 { // for delete endpoints there's no response body
+		return nil
+	}
+
 	// TODO: Make parsing content-type aware. Requires change to go model generation to use interface{} for all union types.
 	// Github Issue: https://github.com/looker-open-source/sdk-codegen/issues/1022
 	switch v := result.(type) {
```

Comment:
Actually, is it possible we can instead check if result is empty or not instead of checking the status code? We by default return nil, we just need to not Decode if its an empty response body. Will make more sense.